repair/repair.hh

```diff
@@ -175,6 +184,7 @@ public:
     int ranges_index = 0;
     repair_stats _stats;
     std::unordered_set<sstring> dropped_tables;
+    abort_source* as;
```

Comment:
This looks pretty risky. The lifetime of node_ops_info (which owns the abort_source this points to) is not synchronized with the lifetime of repair_info. Or is it?

Reply:
It's used only in the constructor to subscribe the callback.
If repair_info is destroyed along with the subscription then it will unsubscribe itself.
There should also be no problem to stop the `sharded<abort_source>` at `node_ops_info` while repair_info is alive since the subscription use boost-intrusive auto_unlink and destroying the list should safely detach the members.

Reply:
We can also go with the approach in https://github.com/bhalevy/scylla/commit/eed170f35e7ea6b3d1d095bf448bb582d7f8845a
where the `repair_info` can be accessed via `node_ops_info` via a foreign shared ptr, so the latter can abort the former using its `abort()` method, similar to the way we plan to do it in the future using the task manager infrastructure.

Reply:
> > It's used only in the constructor to subscribe the callback.
> 
> Then why does it exist as a member of repair_info?
> 
> > If repair_info is destroyed along with the subscription then it will unsubscribe itself.
> > There should also be no problem to stop the sharded<abort_source> at node_ops_info while repair_info is alive since the subscription use boost-intrusive auto_unlink and destroying the list should safely detach the members.
> 
> Absolutely. I've no objection about subscription, it's autounsubscribeable. Just the `abort_source* as` pointer itself

Good point. The member can indeed be removed.